flink-core/src/main/java/org/apache/flink/configuration/TraceOptions.java

```diff
@@ -67,6 +82,31 @@ public static Configuration forReporter(Configuration configuration, String repo
                                     + " any of the names in the list will be started. Otherwise, all reporters that could be found in"
                                     + " the configuration will be started.");
 
+    /** The detail level for reporting checkpoint spans. */
+    public static final ConfigOption<TraceOptions.CheckpointSpanDetailLevel>
+            CHECKPOINT_SPAN_DETAIL_LEVEL =
+                    key("traces.checkpoint.span-detail-level")
+                            .enumType(TraceOptions.CheckpointSpanDetailLevel.class)
+                            .defaultValue(CheckpointSpanDetailLevel.SPAN_PER_CHECKPOINT)
+                            .withDescription(
+                                    String.format(
+                                            "Detail level for reporting checkpoint spans. Possible values:\n"
+                                                    + "- %s (default): Single span per checkpoint. Aggregated sum/max for submetrics from all tasks and subtasks per checkpoint\n"
+                                                    + "- %s: Single span per checkpoint. Same as %s, plus arrays of aggregated values per task.\n"
+                                                    + "- %s: Same as %s plus children spans per each task. Each task span with aggregated sum/max submetrics from subtasks.\n"
+                                                    + "- %s: Same as %s plus children spans per each subtask. Child spans for tasks and grand-child spans for subtasks.",
+                                            CheckpointSpanDetailLevel.SPAN_PER_CHECKPOINT.name(),
+                                            CheckpointSpanDetailLevel.SPAN_PER_CHECKPOINT_WITH_TASKS
+                                                    .name(),
+                                            CheckpointSpanDetailLevel.SPAN_PER_CHECKPOINT.name(),
+                                            CheckpointSpanDetailLevel.CHILDREN_SPANS_PER_TASK
+                                                    .name(),
+                                            CheckpointSpanDetailLevel.SPAN_PER_CHECKPOINT.name(),
+                                            CheckpointSpanDetailLevel.CHILDREN_SPANS_PER_SUBTASK
+                                                    .name(),
+                                            CheckpointSpanDetailLevel.CHILDREN_SPANS_PER_TASK
```

Comment:
Could we use the APIs 'Description.builder().list() ... ' to constructs the description ?
Such as PipelineOptions#SERIALIZATION_CONFIG...

Reply:
Thanks for the pointer, I was not aware of that utility.